cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -241,19 +248,60 @@ compute_submap_indices(const IndexMap& imap,
     std::ranges::sort(global_idx_to_possible_owner);
 
     // Choose the submap owner for each index in `recv_indices`
+    // and pack destination ranks for each process that has received new indices
```

Comment:
So the idea here is that this process (let's call it p) may have received an index (lets call it `i`) that it hasn't been included in the submap, and thus we need to pick the submap owner and tell it what its destinations are for this index. The destination ranks are just the other processes that sent index `i` to this process (`p`). If I've interpreted this correctly, can we maybe improve this comment? I think `received new indices` especially could be made more descriptive

Reply:
Yes, that is the idea.
As we at ownership determination have all ranks that wanted `i`, we know that they will have it in their map, and thus the new owner has these as `dest` ranks.